src/quacc/recipes/espresso/phonons.py

```diff
@@ -0,0 +1,178 @@
+    """
+
+    calc_defaults = {
+        "input_data": {
+            "inputph": {
+                "tr2_ph": 1e-12,
+                "alpha_mix(1)": 0.1,
+                "nmix_ph": 12,
+                "verbosity": "high",
+            }
+        },
+        "qpts": (0, 0, 0),
+    }
+
+    template = EspressoTemplate("ph", test_run=test_run)
+
+    return base_fn(
+        preset=preset,
+        template=template,
+        calc_defaults=calc_defaults,
+        calc_swaps=calc_kwargs,
+        parallel_info=parallel_info,
+        additional_fields={"name": "ph.x Phonon"},
+        copy_files=copy_files,
+    )
+
+
+@flow
+def grid_phonon_flow(
+    custom_pw_job: Job | None = None,
+    custom_phonon_job: Job | None = None,
+    nblocks: int = 1,
+):
+    """
+    Function to carry out a grid parallelization of a ph.x calculation. Each representation of each
+    q-point is calculated in a separate job. This allow the calculations to be run on different machines
+    at possibly different times. The only drawback is that the data of the prior pw.x calculation has to
+    be copied to each of the jobs. Thus the total amount of data will be n*m times the size of the pw.x
+    calculation where n is the number of q-points and m is the number of representations. Plus the data
+    produced by each ph.x calculations. This can becomes very large if your system has a lot of atoms.
+    To provide a way to control the amount of data produced, the user can provide an additional argument
+    "nblocks" which will group multiple representations together in a single job. This will reduce the
+    amount of data produced by a factor of nblocks.
+
+    Parameters
+    ----------
+    custom_pw_job
+        A custom job to run the pw.x calculation. This job should return a RunSchema dictionary.
+        The default job is [quacc.recipes.espresso.core.static_job][].
+    custom_phonon_job
+        A custom job to run the ph.x calculation. This job should return a RunSchema dictionary.
+        The default job is [quacc.recipes.espresso.phonons.phonon_job][].
+    nblocks
+        The number of representations to group together in a single job. This will reduce the amount
+        of data produced by a factor of nblocks. If nblocks = 0 each job will contain all the representations
+        for a single q-point.
+
+    Returns
+    -------
+    RunSchema
+        Dictionary of results from [quacc.schemas.ase.summarize_run][]
+    """
+
+    # We run a first pw job (single point or relax) depending on the input
+    pw_job = static_job if custom_pw_job is None else custom_phonon_job
+    ph_job = phonon_job if custom_phonon_job is None else custom_phonon_job
+
+    pw_job_results = pw_job()
+
+    # First ph job to generate the patterns
+    ph_job_partial = partial(
+        ph_job, test_run=True, copy_files=pw_job_results["dir_name"]
+    )
+    # We actually run the job
+    ph_job_results = ph_job_partial()
```

Comment:
There is no benefit of using a partial function here. The same can be done as:

```python
ph_job_results = ph_job(test_run=True, copy_files=pw_job_results["dir_name"])
```

Reply:
Oh ok, would it still take parameters sent to the original pw_job by the user or override them in this case?

Reply:
If the user supplies a `custom_ph_job` with pre-specified arguments, they will be carried over and not overidden, which is the main purpose of the `partial` function. See an example [here](https://quantum-accelerators.github.io/quacc/user/recipes/recipes_intro.html#modifying-jobs-in-a-dynamic-workflow).